### PR TITLE
Move docs to mkdocs, update user documentation

### PR DESCRIPTION
**Pull request by BSchilperoort**
In this PR the docs are moved to mkdocs, and completely overhauled.

Note: the developer documentation will be added in a later PR.

Todo before merge:
- [ ] Add icon svg file(s)

### CODE REVIEW

**Comment by BSchilperoort** [comment]
@Peter9192 could you take a look at the new docs? I think the user-docs are basically complete.

Note that I still want to add the vector sources for the icons, but Inkscape is quite unstable on my laptop at the moment.

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
```
Comment (Peter9192) [inline docs/CHANGELOG.md:8]: Add changed documentation to changelog?

File: docs/index.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Welcome to era5cli’s documentation!
```
Comment (Peter9192) [inline docs/index.md:1]: Would it be possible to get rid of the `about` header in the left side bar, as it is the same as `home`?

File: docs/getting_started.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
+
+## Installation
+
+For installation, multiple options are available depending on your setup:
+
+=== "Overview"
+
+    | Source | Status |
```
Comment (Peter9192) [inline docs/getting_started.md:8]: I like the content tabs, but I'd limit them purely to the code block. With the surrounding text, it's not clear until where the scope of the tab applies

File: docs/getting_started.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    The development version is available via GitHub. To install directly from GitHub, the following command can be used:
+
+    ```
+    pip install -U  git+https://github.com/eWaterCycle/era5cli.git
+    ```
+
+
+## Copernicus CDS credentials
+
+To be able to use era5cli, you need to be [**registered** at the Copernicus Climate Data Service (CDS)](https://cds.climate.copernicus.eu/user/register?destination=%2F%23!%2Fhome).
```
Comment (Peter9192) [inline docs/getting_started.md:43]: Bold in link suggests there's another role associated with it.

```suggestion
To be able to use era5cli, you need to be [**registered**](https://cds.climate.copernicus.eu/user/register?destination=%2F%23!%2Fhome) at the Copernicus Climate Data Service (CDS).
```

File: docs/getting_started.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+era5cli config --uid ID_NUMBER --key "KEY"
+```
+
+*Where ID_NUMBER is your user ID (e.g. 123456) and "KEY" is your API key, inside double quotes (e.g. "4s215sgs-2dfa-6h34-62h2-1615ad163414").*
+
+After running this command your ID and key are validated and stored inside your home folder, under `.config/era5cli.txt.`
+
+## Your first request
+
+Now you should be able to make a request for data from the CDS. A request you could run can be, for example:
```
Comment (Peter9192) [inline docs/getting_started.md:59]:
```suggestion
Now you should be able to make a request for data from the CDS. For example:
```

File: docs/getting_started.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+```sh
+era5cli monthly --variables soil_type --startyear 2000 --months 01
+```
+
+Which will download the monthly mean `soil_type` data, for January 2000.
+
+*Note: this file is ~2 MB*
```
Comment (Peter9192) [inline docs/getting_started.md:67]: Can we make the footer bar (with navigation to next/previous) blue as well? Or just white (there's another layer of footer at the very bottom, right?). Now, I hardly noticed the navigation there, even though I usually like to use that.

File: docs/formulating_requests.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
+To be able to make a request, you need to decide on a few main points:
+
+- Which variables do you want?
+    - And in which dataset are these variables located? (E.g. ERA5-Land).
+- For which years do you want the variable?
+- Do you want to the data globally, or specify a bounding box (using `--area`).
+- (*For pressure level data:*) on which levels do you want the data.
+
+???+ note
```
Comment (Peter9192) [inline docs/formulating_requests.md:9]: Don't think collapsible is needed for a one-liner

File: docs/formulating_requests.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+
+### Example request
+
+A typical request in `era5cli` can look like the following:
+
+```
+era5cli hourly --land --variables 2m_temperature 2m_dewpoint_temperature
+    --startyear 2000 --endyear 2020 --splitmonths True
+    --area 53.6 3.3 50.7 7.5
```
Comment (Peter9192) [inline docs/formulating_requests.md:21]: Consider one option per line, might be easier for a new user to see what's going on

```suggestion
era5cli hourly \
    --land \
    --variables 2m_temperature 2m_dewpoint_temperature \
    --startyear 2000 \
    --endyear 2020 \
    --splitmonths True \
    --area 53.6 3.3 50.7 7.5
```

File: pyproject.toml [hunk 9]
```diff
@@ -97,7 +104,8 @@ coverage = ["pytest --cov=era5cli --cov-report term --cov-report xml:cov.xml tes
 features = ["docs",]
 
 [tool.hatch.envs.docs.scripts]
-build = ["sphinx-build docs/source docs/build",]
+build = ["mkdocs build",]
```
Comment (Peter9192) [inline pyproject.toml:107]: I get an issue:

```
peter@DESKTOP-S380217:~/ewatercycle/era5cli$ hatch run docs:build
INFO     -  Cleaning site directory
INFO     -  Building documentation to directory: /home/peter/ewatercycle/era5cli/site
Traceback (most recent call last):
  File "/home/peter/.local/share/hatch/env/virtual/era5cli/XytzbWeY/docs/bin/mkdocs", line 8, in <module>
    sys.exit(cli())
  File "/home/peter/.local/share/hatch/env/virtual/era5cli/XytzbWeY/docs/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/peter/.local/share/hatch/env/virtual/era5cli/XytzbWeY/docs/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/peter/.local/share/hatch/env/virtual/era5cli/XytzbWeY/docs/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/peter/.local/share/hatch/env/virtual/era5cli/XytzbWeY/docs/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/peter/.local/share/hatch/env/virtual/era5cli/XytzbWeY/docs/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/peter/.local/share/hatch/env/virtual/era5cli/XytzbWeY/docs/lib/python3.10/site-packages/mkdocs/__main__.py", line 250, in build_command
    build.build(cfg, dirty=not clean)
  File "/home/peter/.local/share/hatch/env/virtual/era5cli/XytzbWeY/docs/lib/python3.10/site-packages/mkdocs/commands/build.py", line 297, in build
    files = config.plugins.run_event('files', files, config=config)
  File "/home/peter/.local/share/hatch/env/virtual/era5cli/XytzbWeY/docs/lib/python3.10/site-packages/mkdocs/plugins.py", line 520, in run_event
    result = method(item, **kwargs)
  File "/home/peter/.local/share/hatch/env/virtual/era5cli/XytzbWeY/docs/lib/python3.10/site-packages/mkdocs_gen_files/plugin.py", line 34, in on_files
    runpy.run_path(file_name)
  File "/usr/lib/python3.10/runpy.py", line 289, in run_path
    return _run_module_code(code, init_globals, run_name,
  File "/usr/lib/python3.10/runpy.py", line 96, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/peter/ewatercycle/era5cli/docs/gen_reference_pages.py", line 77, in <module>
    helpstr = helpstr[helpstr.index("optional arguments") :]
ValueError: substring not found
```

File: docs/formulating_requests.md [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+### Using the info command
+
+To be able to formulate a request, you can make use of the `--help` and `info`
+arguments in era5cli.
+
+The `info` argument takes a name, out of the ones shown below. For example,
+
+```
+era5cli info levels
+```
```
Comment (Peter9192) [inline docs/formulating_requests.md:38]: Somehow I feel these code blocks feel a bit naked. Not sure if this adds anything, but could you try:
```suggestion
```bash
era5cli info levels
```
```

File: docs/reference/variables.md [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+???+ tip
+    With the built-in `grep` command, you can also search through the lists of different variables in era5cli, for example:
+
+    ```
+    era5cli info "2Dvars" | grep temperature
+    ```
+
+    will list all single level variables that contain the word 'temperature'.
+
+### Lists of variables
```
Comment (Peter9192) [inline docs/reference/variables.md:12]: Where does the content come from?

Also: content looks like a code block. Might be helpful to include the command in the code block, e.g. 

```bash
$ era5cli --help
positional arguments:
  {hourly,monthly,info,config}
                        sub-command
```

File: docs/gen_reference_pages.py [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+        "Available pressure levels",
+        "Pressure levels",
+        [str(p) for p in inputref.PLEVELS],
+        4,
+    ),
+    ("Available ERA5-land variables", "ERA5-Land", inputref.ERA5_LAND_VARS, 2),
+]
+
+
+filename = "reference/variables.md"  # Already exists, has the introductory text
```
Comment (Peter9192) [inline docs/gen_reference_pages.py:36]: aha! That's not obvious. Perhaps use https://github.com/ryneeverett/python-markdown-comments to insert an inline comment in the file explaining that info will be added through this script?

File: docs/reference/variables.md (hunk 11 above)
Comment (Peter9192) [reply docs/reference/variables.md:12]: By the way, this is a really nice overview of the available options, and here the content tabs works perfectly!

File: docs/CHANGELOG.md [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+### Changed
+ - Change CDS keys from `.cdsapirc` file to `.config/eracli.txt` file. This will avoid conflict with e.g. ADS.
+ - If a user makes a request without `--splitmonths` they are warned that the behavior will change in the future, and that they have to choose between `--splitmonths False` and `--splitmonths True`.
+ - When a request would encounter a Request Too Large error in the CDS API, they are warned, and given a suggestion to use `--splitmonths`.
+ - `cli.py` has been refactored to make the structure more clear. Seperate argument builders are now in their own modules.
+
+
+## [1.3.2] - 2021-12-13
+### Changed
+ - Elaborate the range of years that can be queried [#123](https://github.com/eWaterCycle/era5cli/pull/123)
```
Comment (Peter9192) [inline docs/CHANGELOG.md:23]: I feel like the sub-menu on the right-hand side of the page has too many levels on the changelog page currently:

image

**Review by Peter9192 — APPROVED** [review]
Very nice work Bart, this documentation is so much better than what we used to have! I love the way you used content tabs for the different options, and that this is generated automatically so we don't have to maintain it in two places. And kudos for the logo!

File: pyproject.toml (hunk 9 above)
Comment (BSchilperoort) [reply pyproject.toml:107]: Oh that's weird. It works fine for me, and on ReadTheDocs as well (the built is successful). I wonder if it is a 3.10 problem, as building the docs is not in the CI.

File: docs/getting_started.md (hunk 6 above)
Comment (BSchilperoort) [reply docs/getting_started.md:67]: Hm that's not very easy, and requires fudging with CSS overrides.

After some searching, I found [some documentation](https://knative.dev/docs/) where they did do that (for the same reason):

https://github.com/knative/docs/blob/d03c86b25d76db58775317b0b86d4fcaa57ef6ea/docs/stylesheets/extra.css#L329-L333

File: docs/formulating_requests.md (hunk 8 above)
Comment (BSchilperoort) [reply docs/formulating_requests.md:21]: Nice, thanks for the suggestion. It's helpful that with the backslash the command is still copy-pasteable

File: docs/formulating_requests.md (hunk 10 above)
Comment (BSchilperoort) [reply docs/formulating_requests.md:38]: That sadly doesn't change anything. It's still colorless

File: docs/getting_started.md (hunk 3 above)
Comment (BSchilperoort) [reply docs/getting_started.md:8]: Ok! I've modified this

File: docs/getting_started.md (hunk 6 above)
Comment (BSchilperoort) [reply docs/getting_started.md:67]: I managed to change it. The color is not quite right for dark mode, but it's perfect on the (default) light mode.

File: pyproject.toml (hunk 9 above)
Comment (BSchilperoort) [reply pyproject.toml:107]: Perhaps your hatch environment is outdated?

**Pull request merged**